src/foam/nanos/notification/email/EmailTemplateEngine.js

```diff
@@ -0,0 +1,425 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.notification.email',
+  name: 'EmailTemplateEngine',
+
+  javaImports: [
+    'foam.dao.DAO',
+    'foam.lib.json.*',
+    'foam.lib.parse.*',
+    'foam.lib.parse.Action',
+    'foam.lib.parse.Grammar',
+    'java.lang.StringBuilder',
+    'java.util.HashMap',
+    'java.util.Map',
+    'static foam.mlang.MLang.EQ'
+  ],
+
+  properties: [
+    {
+      class: 'Map',
+      name: 'values',
+      documentation: `map of values of values for simple values where {{ key }} will be replaced with value`
+    },
+    {
+      class: 'FObjectProperty',
+      of: 'foam.lib.parse.Grammar',
+      name: 'grammar',
+      documentation: `grammar for parsing "{{ val }}", "{% if/else}}" symbols`,
+      javaFactory: `
+        Grammar grammar = new Grammar();
+        grammar.addSymbol("START", grammar.sym("markup"));
+
+        // markup symbol defines the pattern for the whole string
+        grammar.addSymbol("markup", new Repeat0(new Alt(grammar.sym("IF_ELSE"),
+          grammar.sym("IF"), grammar.sym("SIMPLE_VAL"), grammar.sym("ANY_CHAR"))));
+        Action markup = new Action() {
+          @Override
+          public Object execute(Object value, ParserContext x) {
+            return ((StringBuilder) x.get("sb")).toString();
```

Comment:
Do we need to to toString() it? Couldn't we just return the StringBuilder directly? Isn't it just as good as the String itself because they both implement the CharSequence interface.